AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -219,13 +236,19 @@ void CAmSocketHandler::exit_mainloop()
     stop_listening();
 
     //fire the ending filedescriptor
-    int p(1);
-    ssize_t result = write(mPipe[1], &p, sizeof(p));
+    const uint64_t events = UINT64_MAX-1;
+    if (write(mEventFd, &events, sizeof(events)) < 0)
```

Comment:
In theory **write**() is the same as **read**() in terms of not transferring all bytes in one go.  But again, special circumstances might mean this never happens here?

**UPDATE**: Actually I have some vague memory of one Linux page size always being guaranteed and the problem is impossible below one page size...?  But that seems like magical special knowledge...